### DISCUSSION TITLE
Duplicated render with react18+

### DISCUSSION THREAD

**Discussion by NamPNQ** [vercel/next.js#68605] [Help] [answered]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/small-thunder-fjgwc8

### To Reproduce

1. Start application via `next dev`
2. Go to page home `/`

### Current vs. Expected behavior

I expected "Hello" only display one time, or it should throw error if hydration failed

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP PREEMPT_DYNAMIC Sun Aug  6 20:05:33 UTC 2023
  Available memory (MB): 4102
  Available CPU cores: 2
Binaries:
  Node: 20.9.0
  npm: 9.8.1
  Yarn: 1.22.19
  pnpm: 8.10.2
Relevant Packages:
  next: 15.0.0-canary.103 // Latest available version is detected (15.0.0-canary.103).
  eslint-config-next: N/A
  react: 19.0.0-beta-04b058868c-20240508
  react-dom: 19.0.0-beta-04b058868c-20240508
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local)

### Additional context

I tested with next@12 react@17 react-dom@17, it displays "Hello" one time instead two times

**Comment by JesseKoldewijn** [comment, marked as answer]
From what I saw in your sandbox reproduction, this should be the expected behaviour. Since at no time you return two instances of "hello", but rather a different return with the same content.

**Reply by NamPNQ** [reply]
I think the problems is hydrate fail, not about return two instances of hello
For example, if I do 

```
return isServer ? <div>Hello</div> : <div>Hello</div>
```

This also two instance but hydrate success

I just check, this may be new behavior of react 18, not related with nextjs

React 17: https://codesandbox.io/p/sandbox/react-dev-forked-mxr2lc
React 18: https://codesandbox.io/p/sandbox/react-dev-forked-fnzrwr

**Reply by JesseKoldewijn** [reply]
> I think the problems is hydrate fail, not about return two instances of hello
> 
> For example, if I do 
> 
> 
> 
> ```
> 
> return isServer ? <div>Hello</div> : <div>Hello</div>
> 
> ```
> 
> 
> 
> This also two instance but hydrate success
> 
> 
> 
> I just check, this may be new behavior of react 18, not related with nextjs
> 
> 
> 
> React 17: https://codesandbox.io/p/sandbox/react-dev-forked-mxr2lc
> 
> React 18: https://codesandbox.io/p/sandbox/react-dev-forked-fnzrwr

You're actively trying to brute force a hydration error by using a different return per environment.
